service/awsconfig/v6/resource/loadbalancer/current.go

```diff
@@ -44,10 +45,17 @@ func (r *Resource) clusterLoadBalancers(customObject v1alpha1.AWSConfig) (*LoadB
 		allLBNames = append(allLBNames, lb.LoadBalancerName)
 	}
 
-	if len(allLBNames) > 0 {
-		// We get tags for all load balancers which needs a separate API call.
+	// Get loadbalancer tags in batches due to API restriction.
+	for i := 0; i < len(allLBNames); i += loadBalancerTagChunkSize {
```

Comment:
All this is algorithmic and could be extracted in order to have unit tests for it. When this works and is urgent we can merge and sort the problem but I would like to look at this again and make sure we have a way to reliably ensure this is doing what it should. So I think there is some input and some function can apply some magic and generates some output. This is definitely testable. Lets just break down the problem in the code. This will also help understanding how this works.